bytes/concat.ts

```diff
@@ -8,18 +8,32 @@
  * const a = new Uint8Array([0, 1, 2]);
  * const b = new Uint8Array([3, 4, 5]);
  * console.log(concat(a, b)); // [0, 1, 2, 3, 4, 5]
+ * ```
  */
-export function concat(...buf: Uint8Array[]): Uint8Array {
+export function concat(...buf: Uint8Array[] | Uint8Array[][]): Uint8Array {
```

Comment:
This isn't how I would accomplish this overload. I'd probably have two possible signatures: One `buffer: Uint8Array[]` and one `...buf: Uint8Array[]`.

Reply:
Agreed: Check if length is one and then check type:
* Array? Presume array of Uint8Arrays and concat them directly.
* Uint8Array? Return directly as concating a single item requires no copy work.

Reply:
The approach of having such overloads causes signature incompatibility. Do you have an idea how to avoid the conflict?

Also Microsoft recommends using the union as a good practice in case of multiple types: https://www.typescriptlang.org/docs/handbook/2/functions.html#writing-good-overloads

Reply:
Another approach is to copy-paste `concat` into msgpack/encode.ts so it accepts a single argument of type `Uint8Array[]`. 
What do you think?

Reply:
Like this:
```ts
export function encode(object: ValueType) {
  const byteParts: Uint8Array[] = [];
  encodeSlice(object, byteParts);
  
  // Concat byteParts into output 
  let length = 0;
  for (const b of byteParts) {
    length += b.length;
  }
  const output = new Uint8Array(length);
  let index = 0;
  for (const b of byteParts) {
    output.set(b, index);
    index += b.length;
  }
  return output;  
}
```

Reply:
A possible union typing is `Uint8Array[] | [Uint8Array[]]`.

Reply:
@aapoalas implemented per your suggestion together with tests. 
Used `(Uint8Array | Uint8Array[])[]` for better readability and LSP support.